src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/TempPEFileSaveWatcher.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.VS.Automation;
+using VSLangProj;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.TempPE
+{
+    /// <summary>
+    /// This class handles being notified whenever a source file is saved by the hierarchy
+    /// </summary>
+    [Export(typeof(IFileActionHandler))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+    internal class TempPEFileSaveWatcher : IFileActionHandler
+    {
+        private readonly VSBuildManager _buildManager;
+        private readonly Lazy<ITempPEBuildManager> _tempPEBuildManager;
+
+        [ImportingConstructor]
+        public TempPEFileSaveWatcher(
+            [Import(typeof(BuildManager))]VSBuildManager buildManager,
+            Lazy<ITempPEBuildManager> tempPEBuildManager)
+        {
+            _buildManager = buildManager;
+            _tempPEBuildManager = tempPEBuildManager;
+        }
+
+        public async Task<bool> TryHandleFileSavedAsync(IProjectTree tree, string newFilePath, bool saveAs)
+        {
+            // We don't want to be the one to realise the Lazy<ITempPEManager> so ignore changes in projects that don't need TempPE
+            if (!_buildManager.ProjectNeedsTempPE) return false;
+
+            string path = tree.BrowseObjectProperties.ItemName;
```

Comment:
This isn't a path.

Reply:
This feels a lot like bikeshedding, as I suspect if I'd called this `itemName` then it simply would have elicited a comment of "Why are you passing an item name to a method that excepts a path?".

Additionally I think it looks a lot like a path.
![image](https://user-images.githubusercontent.com/754264/49552158-758cf880-f946-11e8-8f40-7f3ece3ddaa5.png)

For reference the above screenshot is from this definition in the project file:
```
    <Compile Include="..\..\OtherResources\Relative\Relative.Designer.cs">
      <AutoGen>True</AutoGen>
      <DesignTime>True</DesignTime>
    </Compile>
```

I'm open to any suggestions anyone might have around this however, I don't feel strongly about it at all.

Reply:
Just because it looks like a path, doesn't make it path. Regardless, I won't dig into this until we've thought about this approach - gone are the days where all manipulation of a file occurs via VS when the code in legacy was originally written. A simply branch switch will cause you to miss a change. I think we want to either watch these files, or have Roslyn figure that out. It already watches all buffers and files on disk.